### PR TITLE
DAOS-4514 debug: do not print the same error

### PR DESCRIPTION
**Pull request by gnailzenh**
dlog could print the same message to stderr for millions
of times if the log file is full.

With this patch, dlog stores the previous error so it can
find the same error and avoid to output message flood

Signed-off-by: Liang Zhen <liang.zhen@intel.com>

### CODE REVIEW

File: src/cart/src/gurt/dlog.c [hunk 1]
```diff
@@ -346,6 +346,25 @@ static void dlog_cleanout(void)
...
+static __thread int	 pre_err_line;
+static __thread uint64_t pre_err_time;
+
+#define dlog_print_err(err, fmt, ...)				\
+do {								\
+	struct timeval	_tv;					\
+								\
+	gettimeofday(&_tv, NULL);				\
+	if (pre_err_line == __LINE__ &&				\
+	    pre_err == (err) &&	_tv.tv_sec <= pre_err_time)	\
```
Comment (NiuYawei) [inline src/cart/src/gurt/dlog.c:359]: The "_tv.tv_sec <= pre_err_time" can never be satisfied, right? It should compare with "pre_err_time + delta"?

Comment (gnailzenh) [reply src/cart/src/gurt/dlog.c:359]: sorry I don't understand, it can be satisfied every second right?
see L361: pre_err_time = _tv.tv_sec;

Comment (NiuYawei) [reply src/cart/src/gurt/dlog.c:359]: yes, I didn't notice it's in sec.

Comment (brianjmurrell) [inline src/cart/src/gurt/dlog.c:359]: Am I understanding correctly that the same error will still be printed once every second?

**Pull request merged**